### PR TITLE
Added SC scraping script

### PR DESCRIPTION
**Pull request by emmanuelgautier1994**

### CODE REVIEW

**Comment by coisnepe** [comment]
Just updated the architecture quite significantly... Could you update this PR accordingly ?
- add the columns to a new migration
- use the new "time 100" scrapper for inspiration. In an nutshell, scrap only the ranking and use the tmdb "normalizer"

**Comment by emmanuelgautier1994** [comment]
> Just updated the architecture quite significantly... Could you update this PR accordingly ?
Done!

> * add the columns to a new migration
Yup!

> * use the new "time 100" scrapper for inspiration. In an nutshell, scrap only the ranking and use the tmdb "normalizer"
Scraping the SC rating separately is WIP because I haven't looked into fetching the movie page URL on SensCritique just yet. Other than that, all good!

> Also, what do you think about using `senscritique` instead of `sc` for column names ?
Works for me!

File: lib/senscritique.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      ranking = movie.css('.elto-rank-item').text.to_i
+      title = movie.css('.elco-original-title').text
+      year = movie.css('.elco-date').text.match(/\d\d\d\d/)[0].to_i
+
+      title_normalized =
+        if title.match(/^(.*?) \(/).nil?
+          title
+        else
+          title.match(/^(.*?) \(/)[1]
+        end
```
Comment (coisnepe) [inline lib/senscritique.rb]: Is this necessary ? From what I can see on their website, `.elco-original-title` has the title and  `.elco-date` has the year?

File: lib/senscritique.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          senscritique_ranking: ranking
+        )
+    end
+  end
+
+  # def self.scrap_movie_rating(movie)
+  #   html = URI.open(movie[:tomato_url])
+  #   doc = Nokogiri.HTML(html)
+  #   doc.css('#tomato_meter_link').text.to_i
+  # end
```
Comment (coisnepe) [inline lib/senscritique.rb]: ✂

File: migrations/6_add_senscritique_111_column.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    add_column :movies, :senscritique_ranking, Integer
```
Comment (coisnepe) [inline migrations/6_add_senscritique_111_column.rb]:
```suggestion
    add_column :movies, :senscritique_ranking, Float
```

File: lib/senscritique.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  def self.scrap_ranking
+    url = 'https://www.senscritique.com/films/tops/top111'
+    html = URI.open(url)
+    doc = Nokogiri.HTML(html)
+    movies_list = doc.css('.elto-item')
+
+    movies_list.each do |movie|
+      ranking = movie.css('.elto-rank-item').text.to_i
+      title = movie.css('.elco-original-title').text
+      year = movie.css('.elco-date').text.match(/\d\d\d\d/)[0].to_i
```
Comment (coisnepe) [inline lib/senscritique.rb]:
```suggestion
      year = movie.css('.elco-date').text.match(/\d\d\d\d/)[0].to_f
```

File: lib/senscritique.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+module SensCritique
+  def self.scrap_ranking
+    url = 'https://www.senscritique.com/films/tops/top111'
+    html = URI.open(url)
+    doc = Nokogiri.HTML(html)
+    movies_list = doc.css('.elto-item')
+
+    movies_list.each do |movie|
+      ranking = movie.css('.elto-rank-item').text.to_i
```
Comment (coisnepe) [inline lib/senscritique.rb]:
```suggestion
      ranking = movie.css('.elto-rank-item').text.to_f
```

File: lib/senscritique.rb (hunk 1 above)
Comment (emmanuelgautier1994) [reply lib/senscritique.rb]: That's an oversight, I copy-pasted the regex and had planned to check what it does but forgot to get back to it 🤦‍♂️Thanks!

File: lib/senscritique.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    end
+  end
+
+  def self.scrap_movie_rating(movie)
+    query_html = URI.open('https://www.senscritique.com/search?q=' + URI.encode(movie[:title]) + '&categories[0][0]=Films')
+    query_doc = Nokogiri.HTML(query_html)
+
+    hits = query_doc.css('.sk-hits').children
+
+    rating = nil
```
Comment (coisnepe) [inline lib/senscritique.rb]: Is this needed? If the condition on l.51 isn't fulfilled, `rating` on l.57 will be `nil` anyway, won't it?

Comment (emmanuelgautier1994) [reply lib/senscritique.rb]: I've tried this with the same thinking and it raises an unknown variable or method error :/

Comment (coisnepe) [reply lib/senscritique.rb]: Absolutely, my bad...

**Comment by coisnepe** [comment]
👌

**Pull request merged**